Automattic/woocommerce-payments
# Modify eslint config and remove camelcase ignores

#### Changes proposed in this Pull Request

Update the eslint camelcase config to still enforce variables and function names a camelCase, but allow for snakecase property names, given our API data is in snake case often.
- Turn off the parameter config for camelcase and also set `ignoreGlobals` on
- Make use of the TypeScript eslint `naming-convention` config instead of eslint camelcase for TypeScript files





#### Testing instructions

- Run `npm run lint`, there should be no errors
- Add the below JS to a JS file:
```
function test_function() {
	var a_variable = 'test';
	return a_variable;
}
```
- Run `npm run lint` and see if camelcase warnings show up for the function name and variable name.
- Do a search through the repo for `camelcase` ignores (there should be none now).



*

-------------------
